src/VisualStudio/Core/Def/Packaging/IPackageSearchIOService.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.LanguageServices.Packaging
+{
+    /// <summary>
+    /// Used so we can mock out how the search service does IO for testing purposes.
+    /// </summary>
+    internal interface IPackageSearchIOService
+    {
+        void Create(DirectoryInfo directory);
+        void Delete(FileInfo file);
+        IEnumerable<FileInfo> EnumerateFiles(DirectoryInfo directory);
+        bool Exists(FileSystemInfo info);
```

Comment:
What would you use instead?  Note: i'm opposed to the pattern of "just try the IO and deal with the excpetion if it happens".  This means exceptions happen all the time.  I don't mind doing an initial peremptory check that will help avoid the exception the majority of the time.  Then, once the call happens, we still do the operation and deal with the exception in (now) much rarer case where there was a race.